File: learning-spring-java/src/main/java/com/ironbird/learningspring/data/ReservationRepository.java
package com.ironbird.learningspring.data;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.repository.CrudRepository;

import java.sql.Date;
import java.util.List;

public interface ReservationRepository extends JpaRepository<Reservation, Long> {
    Iterable<Reservation> findReservationByReservationDate(Date date);
}

File: learning-spring-kotlin/src/main/kotlin/com/ironbird/learningspring/data/ReservationRepository.kt
package com.ironbird.learningspring.data

import org.springframework.data.repository.CrudRepository
import java.sql.Date

interface ReservationRepository : CrudRepository<Reservation, Long> {
    fun findByReservationDate(reservationDate: Date): Iterable<Reservation?>?
    fun findByRoom(room: Room): Iterable<Reservation?>?
}
